MAINT: `cluster`/`stats`: array API test failures in main

### Describe your issue.

The array API CI job is failing (e.g. [here](https://github.com/scipy/scipy/actions/runs/9712970168/job/26808785497)) with these errors, probably due to changes in array-api-strict.

### Reproducing Code Example

```python
NA
```


### Error message

```shell
FAILED scipy/cluster/tests/test_hierarchy.py::TestDendrogram::test_dendrogram_single_linkage_tdist[array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/cluster/tests/test_hierarchy.py::TestDendrogram::test_labels_as_array_or_list[array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/cluster/tests/test_hierarchy.py::TestDendrogram::test_dendrogram_leaf_colors_zero_dist[array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/cluster/tests/test_hierarchy.py::TestDendrogram::test_dendrogram_leaf_colors[array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circfuncs[circvar-0.006455174000787767-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circfuncs[circstd-6.520702116-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circfuncs_small[array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circfuncs_close[circvar-var-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circfuncs_close[circstd-std-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circmean_axis[circvar-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circmean_axis[circstd-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circfuncs_array_like[circvar-0.006455174270186603-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circfuncs_array_like[circstd-6.520702116-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_empty[circvar-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_empty[circstd-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_nan_propagate[circvar-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_nan_propagate[circstd-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_nan_propagate_array[circvar-expected1-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_nan_propagate_array[circstd-expected2-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circfuncs_uint8[array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_morestats.py::TestCircFuncs::test_circstd_zero[array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_resampling.py::TestMonteCarloHypothesisTest::test_input_validation[array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_resampling.py::TestMonteCarloHypothesisTest::test_axis_dtype[-3-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_resampling.py::TestMonteCarloHypothesisTest::test_axis_dtype[-2-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_resampling.py::TestMonteCarloHypothesisTest::test_axis_dtype[-1-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_resampling.py::TestMonteCarloHypothesisTest::test_axis_dtype[0-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_resampling.py::TestMonteCarloHypothesisTest::test_axis_dtype[1-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_resampling.py::TestMonteCarloHypothesisTest::test_axis_dtype[2-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_resampling.py::TestMonteCarloHypothesisTest::test_alternative[two-sided-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_resampling.py::TestMonteCarloHypothesisTest::test_alternative[less-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_resampling.py::TestMonteCarloHypothesisTest::test_alternative[greater-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_r_almost_exactly_pos1[array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_r_almost_exactly_neg1[array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_basic[array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_constant_input[array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_near_constant_input[float32-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_near_constant_input[float64-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_very_small_input_values[array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_very_large_input_values[array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_extremely_large_input_values[array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_length_two_pos1[array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_length_two_neg1[array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_negative_correlation_pvalue_gh17795[array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_length3_r_exactly_negative_one[array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_nd_special_cases[array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_array_api[less-0-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_array_api[less-1-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_array_api[less-None-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_array_api[greater-0-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_array_api[greater-1-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_array_api[greater-None-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_array_api[two-sided-0-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_array_api[two-sided-1-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPearsonr::test_array_api[two-sided-None-array_api_strict] - RuntimeError: The function clip requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestMoments::test_moment_center_array_moment[None-10-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestMoments::test_moment_center_array_moment[None-size1-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestMoments::test_moment_center_array_moment[0-10-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestMoments::test_moment_center_array_moment[0-size1-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestMoments::test_moment_center_array_moment[1-10-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestMoments::test_moment_center_array_moment[1-size1-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestMoments::test_moment[array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat0-case1-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat0-case2-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat0-case3-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat1-case1-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat1-case2-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat1-case3-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat2-case1-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat2-case2-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat2-case3-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat3-case1-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat3-case2-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat3-case3-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat4-case1-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat4-case2-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat4-case3-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat5-case1-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat5-case2-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat5-case3-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat6-case1-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat6-case2-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_basic[lambda_stat6-case3-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_axis[array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_ddof_broadcasting[array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_power_divergence_gh_12282[array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestPowerDivergence::test_power_divergence_against_cressie_read_data[array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestChisquare::test_gh_chisquare_12282[array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestChisquare::test_chiquare_data_types_attributes[200-uint8-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestChisquare::test_chiquare_data_types_attributes[1000000-int32-array_api_strict] - RuntimeError: The function minimum requires API version 2023.12 or later, but the current API version for array-api-strict is 2022.12
FAILED scipy/stats/tests/test_stats.py::TestSkewTest::test_against_R[two-sided-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestSkewTest::test_against_R[less-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestSkewTest::test_against_R[greater-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestKurtosisTest::test_against_R[two-sided-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestKurtosisTest::test_against_R[less-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestKurtosisTest::test_against_R[greater-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestNormalTest::test_against_R[two-sided-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestNormalTest::test_against_R[less-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
FAILED scipy/stats/tests/test_stats.py::TestNormalTest::test_against_R[greater-array_api_strict] - TypeError: array iteration is not allowed in array-api-strict
```


### SciPy/NumPy/Python version and system information

```shell
NA
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -45,7 +45,9 @@ maintainers = [
 # release branches, see:
 #     https://scipy.github.io/devdocs/dev/core-dev/index.html#version-ranges-for-numpy-and-other-dependencies
 requires-python = ">=3.10"
-dependencies = ["numpy>=1.23.5"]  # keep in sync with `min_numpy_version` in meson.build
+dependencies = [
+    "numpy>=1.23.5",
+] # keep in sync with `min_numpy_version` in meson.build
 readme = "README.rst"
 classifiers = [
     "Development Status :: 5 - Production/Stable",
@@ -80,7 +82,7 @@ test = [
     "scikit-umfpack",
     "pooch",
     "hypothesis>=6.30",
-    "array-api-strict",
+    "array-api-strict>=2.0",
     "Cython",
     "meson",
     'ninja; sys_platform != "emscripten"',
@@ -144,13 +146,13 @@ before-build = "bash {project}/tools/wheels/cibw_before_build_linux.sh {project}
 
 [tool.cibuildwheel.linux.environment]
 # /project will be the $PWD equivalent inside the docker used to build the wheel
-PKG_CONFIG_PATH="/project/"
+PKG_CONFIG_PATH = "/project/"
 
 [tool.cibuildwheel.macos]
 before-build = "bash {project}/tools/wheels/cibw_before_build_macos.sh {project}"
 
 [tool.cibuildwheel.macos.environment]
-PKG_CONFIG_PATH="{project}"
+PKG_CONFIG_PATH = "{project}"
 
 [tool.cibuildwheel.windows]
 before-build = "bash {project}/tools/wheels/cibw_before_build_win.sh {project}"
@@ -158,6 +160,6 @@ repair-wheel-command = "bash ./tools/wheels/repair_windows.sh {wheel} {dest_dir}
 
 [tool.cibuildwheel.windows.environment]
 # This does not work because pkg-config does not like backslashes,
-PKG_CONFIG_PATH="{project}"
+PKG_CONFIG_PATH = "{project}"
 # do this instead (which will override this setting)
 # set CIBW_ENVIRONMENT_WINDOWS=PKG_CONFIG_PATH=PWD.replace('\\', '/')

--- a/pytest.ini
+++ b/pytest.ini
@@ -17,3 +17,4 @@ filterwarnings =
     ignore:.*`numpy.core` has been made officially private.*:DeprecationWarning
     ignore:.*In the future `np.long` will be defined as.*:FutureWarning
     ignore:.*JAX is multithreaded.*:RuntimeWarning
+    ignore:.*The 2023.12 version of the array API specification is still preliminary.*:UserWarning

--- a/requirements/test.txt
+++ b/requirements/test.txt
@@ -11,7 +11,7 @@ threadpoolctl
 # scikit-umfpack  # circular dependency issues
 pooch
 hypothesis>=6.30
-array-api-strict
+array-api-strict>=2.0
 Cython
 meson
 ninja; sys_platform != "emscripten"

--- a/scipy/cluster/hierarchy.py
+++ b/scipy/cluster/hierarchy.py
@@ -3321,7 +3321,7 @@ def dendrogram(Z, p=30, truncate_mode=None, color_threshold=None,
 
     if color_threshold is None or (isinstance(color_threshold, str) and
                                    color_threshold == 'default'):
-        color_threshold = max(Z[:, 2]) * 0.7
+        color_threshold = xp.max(Z[:, 2]) * 0.7
 
     R = {'icoord': icoord_list, 'dcoord': dcoord_list, 'ivl': ivl,
          'leaves': lvs, 'color_list': color_list}
@@ -3355,7 +3355,7 @@ def dendrogram(Z, p=30, truncate_mode=None, color_threshold=None,
         above_threshold_color=above_threshold_color)
 
     if not no_plot:
-        mh = max(Z[:, 2])
+        mh = xp.max(Z[:, 2])
         _plot_dendrogram(icoord_list, dcoord_list, ivl, p, n, mh, orientation,
                          no_labels, color_list,
                          leaf_font_size=leaf_font_size,

--- a/scipy/cluster/tests/test_hierarchy.py
+++ b/scipy/cluster/tests/test_hierarchy.py
@@ -948,7 +948,11 @@ class TestDendrogram:
         Z = linkage(xp.asarray(hierarchy_test_data.ytdist), 'single')
         labels = xp.asarray([1, 3, 2, 6, 4, 5])
         result1 = dendrogram(Z, labels=labels, no_plot=True)
-        result2 = dendrogram(Z, labels=list(labels), no_plot=True)
+        # imitating `tolist` in a standard-compliant way
+        labels_list = []
+        for i in range(labels.shape[0]):
+            labels_list += [int(labels[i])]
+        result2 = dendrogram(Z, labels=labels_list, no_plot=True)
         assert result1 == result2
 
     @pytest.mark.skipif(not have_matplotlib, reason="no matplotlib")

--- a/scipy/cluster/tests/test_hierarchy.py
+++ b/scipy/cluster/tests/test_hierarchy.py
@@ -946,13 +946,9 @@ class TestDendrogram:
     def test_labels_as_array_or_list(self, xp):
         # test for gh-12418
         Z = linkage(xp.asarray(hierarchy_test_data.ytdist), 'single')
-        labels = xp.asarray([1, 3, 2, 6, 4, 5])
-        result1 = dendrogram(Z, labels=labels, no_plot=True)
-        # imitating `tolist` in a standard-compliant way
-        labels_list = []
-        for i in range(labels.shape[0]):
-            labels_list += [int(labels[i])]
-        result2 = dendrogram(Z, labels=labels_list, no_plot=True)
+        labels = [1, 3, 2, 6, 4, 5]
+        result1 = dendrogram(Z, labels=xp.asarray(labels), no_plot=True)
+        result2 = dendrogram(Z, labels=labels, no_plot=True)
         assert result1 == result2
 
     @pytest.mark.skipif(not have_matplotlib, reason="no matplotlib")

--- a/scipy/conftest.py
+++ b/scipy/conftest.py
@@ -117,6 +117,9 @@ if SCIPY_ARRAY_API and isinstance(SCIPY_ARRAY_API, str):
     try:
         import array_api_strict
         xp_available_backends.update({'array_api_strict': array_api_strict})
+        array_api_strict.set_array_api_strict_flags(
+            api_version='2023.12'
+        )
     except ImportError:
         pass
 

--- a/scipy/conftest.py
+++ b/scipy/conftest.py
@@ -13,6 +13,7 @@ import hypothesis
 from scipy._lib._fpumode import get_fpu_mode
 from scipy._lib._testutils import FPUModeChangeWarning
 from scipy._lib._array_api import SCIPY_ARRAY_API, SCIPY_DEVICE
+from scipy._lib import _pep440
 
 try:
     from scipy_doctest.conftest import dt_config
@@ -117,6 +118,8 @@ if SCIPY_ARRAY_API and isinstance(SCIPY_ARRAY_API, str):
     try:
         import array_api_strict
         xp_available_backends.update({'array_api_strict': array_api_strict})
+        if _pep440.parse(array_api_strict.__version__) < _pep440.Version('2.0'):
+            raise ImportError("array-api-strict must be >= version 2.0")
         array_api_strict.set_array_api_strict_flags(
             api_version='2023.12'
         )

--- a/scipy/stats/_stats_py.py
+++ b/scipy/stats/_stats_py.py
@@ -1079,11 +1079,12 @@ def moment(a, order=1, axis=0, nan_policy='propagate', *, center=None):
         calculate_mean = center is None and xp.any(order > 1)
         mean = xp.mean(a, axis=axis, keepdims=True) if calculate_mean else None
         mmnt = []
-        for i in order:
-            if center is None and i > 1:
-                mmnt.append(_moment(a, i, axis, mean=mean)[np.newaxis, ...])
+        for i in range(order.shape[0]):
+            order_i = order[i]
+            if center is None and order_i > 1:
+                mmnt.append(_moment(a, order_i, axis, mean=mean)[np.newaxis, ...])
             else:
-                mmnt.append(_moment(a, i, axis, mean=center)[np.newaxis, ...])
+                mmnt.append(_moment(a, order_i, axis, mean=center)[np.newaxis, ...])
         return xp.concat(mmnt, axis=0)
     else:
         return _moment(a, order, axis, mean=center)
@@ -4449,7 +4450,8 @@ def _pearsonr_fisher_ci(r, n, confidence_level, alternative):
         zr = xp.atanh(r)
 
     ones = xp.ones_like(r)
-    n, confidence_level = xp.asarray([n, confidence_level], dtype=r.dtype)
+    n = xp.asarray(n, dtype=r.dtype)
+    confidence_level = xp.asarray(confidence_level, dtype=r.dtype)
     if n > 3:
         se = xp.sqrt(1 / (n - 3))
         if alternative == "two-sided":

--- a/scipy/stats/tests/test_stats.py
+++ b/scipy/stats/tests/test_stats.py
@@ -6397,7 +6397,8 @@ class NormalityTests:
         # test_result@test$p.value
         test_name = self.test_name
         test_fun = getattr(stats, test_name)
-        ref_statistic, ref_pvalue = xp.asarray(self.case_ref)
+        xp_case_ref = xp.asarray(self.case_ref)
+        ref_statistic, ref_pvalue = xp_case_ref[0], xp_case_ref[1]
 
         kwargs = {}
         if alternative in {'less', 'greater'}:

--- a/scipy/stats/tests/test_stats.py
+++ b/scipy/stats/tests/test_stats.py
@@ -6397,8 +6397,8 @@ class NormalityTests:
         # test_result@test$p.value
         test_name = self.test_name
         test_fun = getattr(stats, test_name)
-        xp_case_ref = xp.asarray(self.case_ref)
-        ref_statistic, ref_pvalue = xp_case_ref[0], xp_case_ref[1]
+        ref_statistic= xp.asarray(self.case_ref[0])
+        ref_pvalue = xp.asarray(self.case_ref[1])
 
         kwargs = {}
         if alternative in {'less', 'greater'}:

